### PR TITLE
Feature/support for private

### PR DESCRIPTION
**Pull request by Mihajlo-Pavlovic**

### CODE REVIEW

File: managers/asset-operations-manager.js [hunk 1]
```diff
@@ -293,23 +306,90 @@ export default class AssetOperationsManager {
...
+                continue;
+            }
+            if (privateIndex === privateTripletsSortedAndGrouped.length) {
+                // Only public left no need to do anything
+                break;
+            }
+            const [publicSubject] = publicTripletsSortedAndGrouped[publicIndex][0].split(' ');
+            const [privateSubject] = privateTripletsSortedAndGrouped[privateIndex][0].split(' ');
+            let compare = publicSubject.localeCompare(privateSubject);
+            if (compare === -1) {
```
Comment (brkagithub) [inline managers/asset-operations-manager.js:361]: think it's a bit more natural to do < 0, > 0, and else (equals to 0) for js

File: managers/asset-operations-manager.js [hunk 2]
```diff
@@ -293,23 +306,90 @@ export default class AssetOperationsManager {
...
             dataset = await kcTools.formatDataset(content);
         }
 
-        const numberOfChunks = kcTools.calculateNumberOfChunks(dataset, CHUNK_BYTE_SIZE);
+        dataset.public = kcTools.generateMissingIdsForBlankNodes(dataset.public);
+        if (dataset.private) {
+            dataset.private = kcTools.generateMissingIdsForBlankNodes(dataset.private);
+        }
+
+        let privateTripletsSortedAndGrouped;
```
Comment (brkagithub) [inline managers/asset-operations-manager.js:331]: i think triple is triplet on english, triplets == triples so i would change it like that

File: managers/asset-operations-manager.js [hunk 3]
```diff
@@ -293,23 +306,90 @@ export default class AssetOperationsManager {
...
+            if (privateIndex === privateTripletsSortedAndGrouped.length) {
+                // Only public left no need to do anything
+                break;
+            }
+            const [publicSubject] = publicTripletsSortedAndGrouped[publicIndex][0].split(' ');
+            const [privateSubject] = privateTripletsSortedAndGrouped[privateIndex][0].split(' ');
+            let compare = publicSubject.localeCompare(privateSubject);
+            if (compare === -1) {
+                // Public comes next
+                publicIndex += 1;
```
Comment (brkagithub) [inline managers/asset-operations-manager.js:363]: you have public index in all parts of if statement, you can put it outside of it probably

File: managers/asset-operations-manager.js [hunk 4]
```diff
@@ -248,6 +249,18 @@ export default class AssetOperationsManager {
         return BigInt(allowance);
     }
 
+    /**
```
Comment (brkagithub) [inline managers/asset-operations-manager.js:252]: maybe it can go to utilities instead of here

File: managers/asset-operations-manager.js (hunk 1 above)
Comment (Mihajlo-Pavlovic) [reply managers/asset-operations-manager.js]: Done added some coments and a bit refactor to make it easier to follow

File: managers/asset-operations-manager.js (hunk 3 above)
Comment (Mihajlo-Pavlovic) [reply managers/asset-operations-manager.js:374]: That's correct, but I think this is easier to read so you have logic for each case there in that block

**Comment by brkagithub** [comment]
https://github.com/OriginTrail/dkg.js/pull/186 <- added here

**Pull request closed without merging**